metrics/metrics.go

```diff
@@ -0,0 +1,40 @@
+package metrics
+
+import (
+	"sort"
+	"sync"
+)
+
+type (
+	FileMetrics struct {
+		ValidEntries   int32
+		InvalidEntries int32
+		ValidBytes     int64
+		InvalidBytes   int64
+	}
+	dbFileMetrics map[int32]FileMetrics
+)
+
+var (
+	once sync.Once
+	dbfm dbFileMetrics
+)
+
+func Init()  { once.Do(func() { dbfm = make(dbFileMetrics) }) }
+func reset() { dbfm = make(dbFileMetrics) }
+
+func DeleteMetrics(fd int)                   { delete(dbfm, int32(fd)) }
```

Comment:
It is recommended that the code style be consistent.
Suggest this instead of writing it in one line:
```go
func foo() {
...
}
```